wandb/sdk/wandb_init.py

```diff
@@ -299,9 +302,11 @@ def init(self):
         trigger.call("on_init", **self.kwargs)
         s = self.settings
         config = self.config
-
         if s._noop:
-            run = RunDummy()
+            run = Dummy()
+            run.config = config
+            run.summary = dict()
+            run.log = lambda data: run.summary.update(data)
```

Comment:
@vanpelt @raubitsj Does this look reasonable?

Reply:
`log` accepts additional optional keyword arguments.  You can probably just ignore them, but we may only want to update summary using the same logic as `commit=True || step > self._step`.  This won't add _runtime, or _step to summary which it's possible some users would expect, probably ok to skip for now.

I would confirm the `config` and `summary` object we're using here don't assume a backend is running or write any data to disk.

Reply:
Also, `summary` and `config` are actually fancier than dicts.  We should likely create a `DummyDict` class that allows users to access these objects like `summary.foo` and `summary.foo.update({"bar": 1})`.

Reply:
We might be able to just the actual Summary and Config objects and disable their communication with the backend or file system.